test/device/add_inlink_test.cpp

```diff
@@ -0,0 +1,28 @@
+#include <gtest/gtest.h>
+
+#include "utils.hpp"
+
+namespace {
+
+class AddInlink : public testing::Test {
+public:
+    void TearDown() override{};
+    void SetUp() override{};
+};
+
+TEST_F(AddInlink, LinkPresents) {
+    sim::TestDevice* source = new sim::TestDevice();
+    sim::TestDevice* dest = new sim::TestDevice();
+    sim::TestLink* link1 = new sim::TestLink(source, dest);
+    sim::TestLink* link2 = new sim::TestLink(source, dest);
```

Comment:
You creates this objects, but does not destroy it. But instead of destructing you may locate them on stack:
 ```suggestion
    sim::TestDevice source = sim::TestDevice();
    sim::TestDevice dest = sim::TestDevice();
    sim::TestLink link1 = sim::TestLink(source, dest);
    sim::TestLink link2 = sim::TestLink(source, dest);
```